### ISSUE TITLE
(Wireguard) tcpdump shows packets coming but connected devices get no internet access

### ISSUE THREAD

**Issue by Raizelian** [pivpn/pivpn#1523]
## In raising this issue, I confirm the following:

- [X] I have read the [documentation](https://docs.pivpn.io)
- [ ] Is it a feature request? please consider opening a [Discussion] (https://github.com/pivpn/pivpn/discussions/new)
- [X] I have read and understood the [contributors guide](https://github.com/pivpn/pivpn/blob/master/CONTRIBUTING.md).
- [ ] The issue I am reporting can be *replicated*.  
- [ ] The issue I am reporting *is* directly related to the pivpn installer script.
- [X] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pivpn/pivpn/wiki/FAQ), [closed issues](https://github.com/pivpn/pivpn/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aclosed), and [open issues](https://github.com/pivpn/pivpn/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aopen)).

### Has your install failed? 
```
yes/[no]
```

### Describe the issue
I installed pivpn (testing branch*) on a fresh Raspberry Pi Os installation; the install ends successfully and I am able to add a client and generate a qr code for it (an iPhone, in my case). Once I connect to the VPN, however, I am unable to access my network.
I used an app to ping my gateway, my Raspberry Pi and any website, but nothing goes through.

*Note: I installed pivpn from the testing branch because I was getting errors about setting up a static IP address when installing normally, and I read in other issues that this could be a solution (and it is).

### Have you searched for similar issues and solutions?
    ([yes]/no / which issues?)

### Have you taken any steps towards solving your issue?
```
I tried uninstalling and reinstalling everything a few times and then I tried with different settings (e.g. different ports, different DNS, no DDNS; empty “Allowed IPs” field or “Exclude private IPs” enabled on the client) but nothing works. These are my first attempts at installing a VPN on a Raspberry Pi, since the device itself is new, so I could be missing something obvious.

`pivpn -c` shows the client as connected.

I followed the FAQs on docs.pivpn.io and every step checks out, including the tcpdump (at least as far as I’m understanding it?).
Here is the output of that command, as root:

> tcpdump -n -i eth0 udp port 11940
tcpdump: verbose output suppressed, use -v[v]... for full protocol decode
listening on eth0, link-type EN10MB (Ethernet), snapshot length 262144 bytes
13:41:18.362390 IP REDACTED > 192.168.1.18.11940: UDP, length 96
13:41:18.362398 IP REDACTED > 192.168.1.18.11940: UDP, length 96
13:41:18.362631 IP REDACTED > 192.168.1.18.11940: UDP, length 96
13:41:18.362636 IP REDACTED > 192.168.1.18.11940: UDP, length 96
13:41:18.362639 IP REDACTED > 192.168.1.18.11940: UDP, length 112
13:41:18.362643 IP REDACTED > 192.168.1.18.11940: UDP, length 112
13:41:18.362646 IP REDACTED > 192.168.1.18.11940: UDP, length 96
13:41:18.362649 IP REDACTED > 192.168.1.18.11940: UDP, length 96
13:41:18.362652 IP REDACTED > 192.168.1.18.11940: UDP, length 96
13:41:18.362818 IP REDACTED > 192.168.1.18.11940: UDP, length 96
13:41:18.362823 IP REDACTED > 192.168.1.18.11940: UDP, length 96
13:41:18.366246 IP 192.168.1.18.11940 > REDACTED: UDP, length 112
13:41:18.368603 IP 192.168.1.18.11940 > REDACTED: UDP, length 352
13:41:18.373159 IP 192.168.1.18.11940 > REDACTED: UDP, length 192
13:41:18.374601 IP 192.168.1.18.11940 > REDACTED: UDP, length 160
13:41:18.375348 IP 192.168.1.18.11940 > REDACTED: UDP, length 288
13:41:18.376095 IP 192.168.1.18.11940 > REDACTED: UDP, length 352
13:41:18.378511 IP 192.168.1.18.11940 > REDACTED: UDP, length 112
13:41:18.378966 IP 192.168.1.18.11940 > REDACTED: UDP, length 224
13:41:18.379449 IP 192.168.1.18.11940 > REDACTED: UDP, length 208
13:41:18.380209 IP 192.168.1.18.11940 > REDACTED: UDP, length 224
13:41:18.381062 IP 192.168.1.18.11940 > REDACTED: UDP, length 352
13:41:20.374367 IP  REDACTED > 192.168.1.18.11940: UDP, length 96
13:41:20.374373 IP REDACTED > 192.168.1.18.11940: UDP, length 96
(…)
^C
59 packets captured
59 packets received by filter
0 packets dropped by kernel
```

#### What type of hardware are you running PiVPN at?

```
Raspberry Pi 4 Model B
```
#### Output of `uname -a`
```
Linux nespi 5.15.32-v8+ #1538 SMP PREEMPT Thu Mar 31 19:40:39 BST 2022 aarch64 GNU/Linux
```

#### Output of `cat /etc/os-release`
```
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

### If install failed Please provide the console output of  `curl -L https://install.pivpn.io | bash`
```
  install did not fail
```

### Console output of      `curl -L install.pivpn.io | bash`
```
% Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    17  100    17    0     0     96      0 --:--:-- --:--:-- --:--:--    97
100   162  100   162    0     0    375      0 --:--:-- --:--:-- --:--:--   375
[100 92976](tel:100 92976)  [100 92976](tel:100 92976)    0     0   106k      0 --:--:-- --:--:-- --:--:--  524k
:::
::: sudo will be used for the install.
::: Repair option selected.
::: Hostname length OK
::: Verifying free disk space...
:::
::: Package Cache update is needed, running apt-get update -y ...
 done!
:::
::: Checking apt-get for upgraded packages.... done!
:::
::: Your system is up to date! Continuing with PiVPN installation...
:::    Checking for git... already installed!
:::    Checking for tar... already installed!
:::    Checking for curl... already installed!
:::    Checking for grep... already installed!
:::    Checking for dnsutils... already installed!
:::    Checking for grepcidr... already installed!
:::    Checking for whiptail... already installed!
:::    Checking for net-tools... already installed!
:::    Checking for bsdmainutils... already installed!
:::    Checking for bash-completion... already installed!
:::    Checking for iptables-persistent... already installed!
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
::: PiVPN Automated Installer
::: This installer will transform your Debian host into an OpenVPN or WireGuard server!
::: Initiating network interface
::: Enable forced IPv6 route with no IPv6 uplink on server.
::: Using interface: eth0 for IPv4
::: Since we think you are not using Raspbian, we will not configure a static IP for you.
::: nes will hold your ovpn configurations.
:::
::: Checking for existing base files...
:::    Checking /usr/local/src/pivpn is a repo... OK!
::: Repairing an existing installation, not downloading/updating local repos
::: WireGuard will be installed
::: Installing WireGuard from Debian package... 
:::    Checking for wireguard-tools... already installed!
:::    Checking for qrencode... already installed!
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
::: Using port 11940
::: Using DNS 192.168.1.1 8.8.8.8
dig failed, now trying to curl checkip.amazonaws.com
::: Using domain name REDACTED
::: Backing up the wireguard folder to /etc/wireguard_2022-05-13-135142.tar.gz
::: The Server Keys will now be generated.
::: Server Keys have been generated.
::: Server config generated.
::: Install Complete...
::: Restarting services...
::: Enabling unattended upgrades
:::    Checking for unattended-upgrades... already installed!
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
::: Setupfiles copied to /etc/pivpn/wireguard/setupVars.conf
::: Installing scripts to /opt/pivpn...
 done.
::: Flushing writes to disk...
::: done.
::: Installation Complete!
::: Now run 'pivpn add' to create the client profiles.
::: Run 'pivpn help' to see what else you can do!

::: If you run into any issue, please read all our documentation carefully.
::: All incomplete posts or bug reports will be ignored or deleted.

::: Thank you for using PiVPN.
::: It is strongly recommended you reboot after installation.
:::
```

### Console output of      `pivpn debug`

```
::: Generating Debug Output
::::            PiVPN debug              ::::
=============================================
::::            Latest commit            ::::
Branch: test
Commit: a5e41c466fd9539dcbfaaee4ca13853fca87e107
Author: Orazio
Date: Tue May 3 17:14:31 2022 +0200
Summary: Use easy-rsa commit 8d1be1c instead of release 3.0.7 for OpenSSL 3 support.
=============================================
::::        Installation settings        ::::
PLAT=Debian
OSCN=bullseye
USING_UFW=0
pivpnforceipv6route=1
IPv4dev=eth0
install_user=nes
install_home=/home/nes
VPN=wireguard
pivpnPORT=11940
pivpnDNS1=192.168.1.1
pivpnDNS2=8.8.8.8
pivpnHOST=REDACTED
INPUT_CHAIN_EDITED=0
FORWARD_CHAIN_EDITED=1
INPUT_CHAIN_EDITEDv6=
FORWARD_CHAIN_EDITEDv6=
pivpnPROTO=udp
pivpnMTU=1420
pivpnDEV=wg0
pivpnNET=10.23.80.0
subnetClass=24
pivpnenableipv6=0
ALLOWED_IPS="0.0.0.0/0, ::0/0"
UNATTUPG=1
INSTALLED_PACKAGES=(dnsutils grepcidr bsdmainutils iptables-persistent wireguard-tools qrencode unattended-upgrades)
=============================================
::::  Server configuration shown below   ::::
[Interface]
PrivateKey = server_priv
Address = 10.23.80.1/24
MTU = 1420
ListenPort = 11940
### begin test ###
[Peer]
PublicKey = test_pub
PresharedKey = test_psk
AllowedIPs = 10.23.80.2/32
### end test ###
=============================================
::::  Client configuration shown below   ::::
[Interface]
PrivateKey = test_priv
Address = 10.23.80.2/24
DNS = 192.168.1.1, 8.8.8.8

[Peer]
PublicKey = server_pub
PresharedKey = test_psk
Endpoint = REDACTED:11940
AllowedIPs = 0.0.0.0/0, ::0/0
=============================================
::::    Recursive list of files in       ::::
::::    /etc/wireguard shown below       ::::
/etc/wireguard:
configs
keys
wg0.conf

/etc/wireguard/configs:
clients.txt
test.conf

/etc/wireguard/keys:
server_priv
server_pub
test_priv
test_psk
test_pub
=============================================
::::            Self check               ::::
:: [OK] IP forwarding is enabled
:: [OK] Iptables MASQUERADE rule set
:: [OK] Iptables FORWARD rule set
:: [OK] WireGuard is running
:: [OK] WireGuard is enabled (it will automatically start on reboot)
:: [OK] WireGuard is listening on port 11940/udp
=============================================
:::: Having trouble connecting? Take a look at the FAQ:
:::: https://docs.pivpn.io/faq
=============================================
:::: WARNING: This script should have automatically masked sensitive       ::::
:::: information, however, still make sure that PrivateKey, PublicKey      ::::
:::: and PresharedKey are masked before reporting an issue. An example key ::::
:::: that you should NOT see in this log looks like this:                  ::::
:::: YIAoJVsdIeyvXfGGDDadHh6AxsMRymZTnnzZoAb9cxRe                          ::::
=============================================
::::            Debug complete           ::::
::: 
::: Debug output completed above.
::: Copy saved to /tmp/debug.log
::: 
```

**Comment by orazioedoardo** [comment]
I would say DNS, but since not even ping works, I think the Raspberry Pi is not forwarding the traffic, so post the output of `sudo iptables -vnL FORWARD`

**Comment by Raizelian** [comment]
Thanks for answering, here’s the output:

`Chain FORWARD (policy DROP 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination         
    0     0 DOCKER-USER  all  --  *      *       0.0.0.0/0            0.0.0.0/0           
    0     0 DOCKER-ISOLATION-STAGE-1  all  --  *      *       0.0.0.0/0            0.0.0.0/0           
    0     0 ACCEPT     all  --  *      docker0  0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED
    0     0 DOCKER     all  --  *      docker0  0.0.0.0/0            0.0.0.0/0           
    0     0 ACCEPT     all  --  docker0 !docker0  0.0.0.0/0            0.0.0.0/0           
    0     0 ACCEPT     all  --  docker0 docker0  0.0.0.0/0            0.0.0.0/0           
    0     0 ACCEPT     all  --  eth0   wg0     0.0.0.0/0            10.23.80.0/24        ctstate RELATED,ESTABLISHED /* wireguard-forward-rule */
    0     0 ACCEPT     all  --  wg0    eth0    10.23.80.0/24        0.0.0.0/0            /* wireguard-forward-rule */ `

**Comment by orazioedoardo** [comment]
It may be that the `DOCKER-USER` and `DOCKER-ISOLATION-STAGE-1` chains are dropping packets. The easiest way to fix is to delete the wireguard rules and insert them back to the top to maximize the priority:


Delete:
```
sudo iptables -D FORWARD -d 10.23.80.0/24 -i eth0 -o wg0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT -m comment --comment "wireguard-forward-rule"
sudo iptables -D FORWARD -s 10.23.80.0/24 -i wg0 -o eth0 -j ACCEPT -m comment --comment "wireguard-forward-rule"
```

Insert:
```
sudo iptables -I FORWARD 1 -d 10.23.80.0/24 -i eth0 -o wg0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT -m comment --comment "wireguard-forward-rule"
sudo iptables -I FORWARD 2 -s 10.23.80.0/24 -i wg0 -o eth0 -j ACCEPT -m comment --comment "wireguard-forward-rule"
```

**Comment by Raizelian** [comment]
Done. Running the previous command shows the changed output (those two lines are now in first and second position) but I still get no internet access (tried pinging, browsing and all that).

**Comment by orazioedoardo** [comment]
Can you try tcpdump over the virtual infervace while pinging `192.168.1.1`? `sudo tcpdump -n -i wg0`

**Comment by Raizelian** [comment]
Yes I was pinging from my configured client (the iPhone) and those were the outputs of the tcpdump command on the Pi.

Also actually, in the meantime I was cleaning up some stuff on the Pi and looking up other things and while I was doing that the VPN actually worked for a minute or so lol
I was able to ping my gateway and everything else from my phone, and when I looked up my public IP on a generic website it was the same public IP that my ethernet-connected home computer gets, not the one I get on mobile data. 

Temporary success!
However, after a minute or so of me gloating it's back to not working again. I'm not sure what's happening here.
Visiting 192.168.1.1 from the client is back to not working at all, too.

**Comment by orazioedoardo** [comment]
> Yes I was pinging from my configured client (the iPhone) and those were the outputs of the tcpdump command on the Pi.

The tcpdump log is showing the following:

`18:02:37.426851 IP 10.23.80.1> 10.23.80.1: ICMP echo request, id 12992, seq 2, length 64`

Which means that 10.23.80.1 (the virtual IP of the raspberry pi) is pinging 10.23.80.1, while you iphone is 10.23.80.2, assuming it is _test_ in the debug log.

> However, after a minute or so of me gloating it's back to not working again.

It could be that the NAT session on the router is expiring and the iphone is not doing the handshake anymore, but that would be a bug in the Apple VPN framework or wireguard, can you check wireguard logs on the iphone (they are in the app)?

>  I was trying stuff in the wg0.conf and added a PersistentKeepalive=25 option in the Peer settings of my client and it looks like it worked for a moment after that

Usually this is not needed for a roaming client, since it will just perform the handshake on demand, and keeping a connection open uses more battery.

If you just turn off and turn back on the VPN when it stops working, does it start working again?

**Comment by Raizelian** [comment]
> Which means that 10.23.80.1 (the virtual IP of the raspberry pi) is pinging 10.23.80.1, while you iphone is 10.23.80.2, assuming it is test in the debug log.

I'm not sure what happened there to be honest, because I was pinging from the app on my phone, that yes, is called "test" in the debug log.
I've tried the command once more, not sure if it's still relevant but here are the results:
```
tcpdump: verbose output suppressed, use -v[v]... for full protocol decode
listening on wg0, link-type RAW (Raw IP), snapshot length 262144 bytes
19:02:46.541718 IP 10.23.80.2 > 10.23.80.1: ICMP echo request, id 18867, seq 7, length 64
19:02:46.541823 IP 10.23.80.1 > 10.23.80.2: ICMP echo reply, id 18867, seq 7, length 64
19:02:46.812200 IP 10.23.80.2.53077 > 8.8.4.4.53: 38149+ Type65? gateway.icloud.com. (36)
19:02:46.812398 IP 10.23.80.2.51746 > 8.8.4.4.53: 29397+ A? gateway.icloud.com. (36)
19:02:46.823461 IP 8.8.4.4.53 > 10.23.80.2.53077: 38149 1/1/0 CNAME gateway.fe.apple-dns.net. (144)
19:02:46.824138 IP 8.8.4.4.53 > 10.23.80.2.51746: 29397 9/0/0 CNAME gateway.fe.apple-dns.net., A 17.248.145.175, A 17.248.248.81, A 17.248.248.22, A 17.248.145.236, A 17.248.145.210, A 17.248.145.103, A 17.248.248.8, A 17.248.145.75 (202)
19:02:47.551475 IP 10.23.80.2 > 10.23.80.1: ICMP echo request, id 18867, seq 8, length 64
19:02:47.551570 IP 10.23.80.1 > 10.23.80.2: ICMP echo reply, id 18867, seq 8, length 64
19:02:47.731619 IP 10.23.80.2.63221 > 8.8.8.8.53: 63490+ A? apple.com. (27)
19:02:47.731752 IP 10.23.80.2.60967 > 8.8.8.8.53: 21707+ Type65? securepubads.g.doubleclick.net. (48)
19:02:47.732545 IP 8.8.8.8.53 > 10.23.80.2.63221: 63490 1/0/0 A 17.253.144.10 (43)
19:02:47.738946 IP 10.23.80.2.49558 > 8.8.8.8.53: 37902+ A? securepubads.g.doubleclick.net. (48)
19:02:47.738946 IP 10.23.80.2.64602 > 8.8.8.8.53: 53645+ Type65? f0480d9778d90f89c600291bdc493650.safeframe.googlesyndication.com. (82)
19:02:47.739116 IP 10.23.80.2.59283 > 8.8.8.8.53: 31762+ A? f0480d9778d90f89c600291bdc493650.safeframe.googlesyndication.com. (82)
19:02:47.739116 IP 10.23.80.2.50467 > 8.8.8.8.53: 15299+ Type65? outlook.live.com. (34)
19:02:47.739116 IP 10.23.80.2.51633 > 8.8.8.8.53: 3769+ A? outlook.live.com. (34)
19:02:47.739116 IP 10.23.80.2.62617 > 8.8.8.8.53: 15561+ Type65? www.wired.it. (30)
19:02:47.739410 IP 10.23.80.2.53007 > 8.8.8.8.53: 64291+ A? www.wired.it. (30)
19:02:47.739410 IP 10.23.80.2.64767 > 8.8.8.8.53: 43584+ Type65? www.google.it. (31)
19:02:47.739411 IP 10.23.80.2.49628 > 8.8.8.8.53: 48683+ A? www.google.it. (31)
19:02:47.739411 IP 10.23.80.2.57567 > 8.8.8.8.53: 44318+ Type65? in.appcenter.ms. (33)
19:02:47.739411 IP 10.23.80.2.54982 > 8.8.8.8.53: 54267+ A? in.appcenter.ms. (33)
19:02:47.739411 IP 10.23.80.2.50655 > 8.8.8.8.53: 10890+ A? www.google.com. (32)
19:02:47.739411 IP 10.23.80.2.62145 > 8.8.8.8.53: 30819+ Type65? googleads.g.doubleclick.net. (45)
19:02:47.739411 IP 10.23.80.2.53005 > 8.8.8.8.53: 25634+ A? googleads.g.doubleclick.net. (45)
19:02:47.739733 IP 10.23.80.2.56102 > 8.8.8.8.53: 52826+ A? app-measurement.com. (37)
19:02:47.739733 IP 10.23.80.2.52537 > 8.8.8.8.53: 62543+ A? https://app-measurement.com/sdk-exp. (53)
19:02:47.739734 IP 10.23.80.2.59639 > 8.8.8.8.53: 7597+ Type65? pagead2.googlesyndication.com. (47)
19:02:47.739734 IP 10.23.80.2.59968 > 8.8.8.8.53: 25427+ A? pagead2.googlesyndication.com. (47)
19:02:47.739734 IP 10.23.80.2.59366 > 8.8.8.8.53: 2062+ Type65? api.techet.net. (32)
19:02:47.739734 IP 10.23.80.2.51537 > 8.8.8.8.53: 27544+ A? api.techet.net. (32)
19:02:47.739914 IP 8.8.8.8.53 > 10.23.80.2.49558: 37902 2/0/0 CNAME partnerad.l.doubleclick.net., A 142.250.184.66 (105)
19:02:47.741188 IP 8.8.8.8.53 > 10.23.80.2.59283: 31762 2/0/0 CNAME pagead-googlehosted.l.google.com., A 142.251.209.1 (144)
19:02:47.742381 IP 8.8.8.8.53 > 10.23.80.2.60967: 21707 1/1/0 CNAME partnerad.l.doubleclick.net. (134)
19:02:47.749918 IP 8.8.8.8.53 > 10.23.80.2.64602: 53645 1/1/0 CNAME pagead-googlehosted.l.google.com. (175)
19:02:47.750137 IP 8.8.8.8.53 > 10.23.80.2.49628: 48683 1/0/0 A 142.250.180.163 (47)
19:02:47.751293 IP 8.8.8.8.53 > 10.23.80.2.50467: 15299 4/1/0 CNAME outlook.office365.com., CNAME outlook.ha.office365.com., CNAME outlook.ms-acdc.office.com., CNAME MRS-efz.ms-acdc.office.com. (219)
19:02:47.752442 IP 8.8.8.8.53 > 10.23.80.2.51633: 3769 8/0/0 CNAME outlook.office365.com., CNAME outlook.ha.office365.com., CNAME outlook.ms-acdc.office.com., CNAME MRS-efz.ms-acdc.office.com., A 40.101.93.226, A 52.97.168.194, A 52.97.173.2, A 52.98.200.194 (214)
19:02:47.752674 IP 8.8.8.8.53 > 10.23.80.2.50655: 10890 1/0/0 A 142.250.184.100 (48)
19:02:47.752981 IP 8.8.8.8.53 > 10.23.80.2.62617: 15561 1/1/0 CNAME cni-digital.map.fastly.net. (131)
19:02:47.754153 IP 8.8.8.8.53 > 10.23.80.2.53007: 64291 2/0/0 CNAME cni-digital.map.fastly.net., A 151.101.242.133 (86)
19:02:47.754357 IP 8.8.8.8.53 > 10.23.80.2.53005: 25634 1/0/0 A 142.251.209.2 (61)
19:02:47.754608 IP 8.8.8.8.53 > 10.23.80.2.56102: 52826 1/0/0 A 142.250.180.174 (53)
19:02:47.756468 IP 8.8.8.8.53 > 10.23.80.2.59968: 25427 1/0/0 A 142.250.180.130 (63)
19:02:47.757565 IP 8.8.8.8.53 > 10.23.80.2.51537: 27544 2/0/0 CNAME ghs.googlehosted.com., A 142.250.184.83 (82)
19:02:47.777146 IP 8.8.8.8.53 > 10.23.80.2.62145: 30819 0/1/0 (105)
19:02:47.777292 IP 8.8.8.8.53 > 10.23.80.2.57567: 44318 2/1/0 CNAME in2-prod-east-us2-23fa330.trafficmanager.net., CNAME in2-gw2-01-ce7dd027.eastus2.cloudapp.azure.com. (221)
19:02:47.777473 IP 8.8.8.8.53 > 10.23.80.2.54982: 54267 3/0/0 CNAME in2-prod-east-us2-23fa330.trafficmanager.net., CNAME in2-gw2-01-ce7dd027.eastus2.cloudapp.azure.com., A 52.177.138.113 (167)
19:02:47.777658 IP 8.8.8.8.53 > 10.23.80.2.52537: 62543 NXDomain 0/1/0 (128)
19:02:47.777803 IP 8.8.8.8.53 > 10.23.80.2.59639: 7597 0/1/0 (104)
19:02:47.778370 IP 8.8.8.8.53 > 10.23.80.2.64767: 43584 0/1/0 (91)
19:02:47.778496 IP 8.8.8.8.53 > 10.23.80.2.59366: 2062 1/1/0 CNAME ghs.googlehosted.com. (123)
19:02:47.871524 IP 10.23.80.2.61796 > 8.8.4.4.53: 41810+ PTR? 1.65.128.10.in-addr.arpa. (42)
19:02:47.881875 IP 8.8.4.4.53 > 10.23.80.2.61796: 41810 NXDomain 0/1/0 (119)
^C
54 packets captured
54 packets received by filter
0 packets dropped by kernel
```



> It could be that the NAT session on the router is expiring and the iphone is not doing the handshake anymore, but that would be a bug in the Apple VPN framework or wireguard, can you check wireguard logs on the iphone (they are in the app)?

Luckily I took a screenshot at the time the VPN was working, so here are the Wireguard logs at that exact time:
```
(...)
2022-05-13 18:19:33.003512: [NET] peer(CsqG…0myU) - Sending handshake initiation
2022-05-13 18:19:38.179102: [NET] peer(CsqG…0myU) - Handshake did not complete after 5 seconds, retrying (try 2)
2022-05-13 18:19:38.179388: [NET] peer(CsqG…0myU) - Sending handshake initiation
2022-05-13 18:19:38.228227: [NET] peer(CsqG…0myU) - Received handshake response
2022-05-13 18:19:38.228460: [NET] peer(CsqG…0myU) - Sending keepalive packet
2022-05-13 18:21:55.748051: [NET] peer(CsqG…0myU) - Sending handshake initiation
2022-05-13 18:21:56.092121: [NET] peer(CsqG…0myU) - Received handshake response
2022-05-13 18:21:56.092313: [NET] peer(CsqG…0myU) - Sending keepalive packet

2022-05-13 18:21:56.840635: [NET] Network change detected with satisfied route and interface order [pdp_ip0, utun2]
2022-05-13 18:21:56.841839: [NET] DNS64: mapped 93.149.216.183 to itself.
2022-05-13 18:21:56.842157: [NET] peer(CsqG…0myU) - UAPI: Updating endpoint
2022-05-13 18:21:56.842386: [NET] Routine: receive incoming v4 - stopped
2022-05-13 18:21:56.842427: [NET] Routine: receive incoming v6 - stopped
2022-05-13 18:21:56.842692: [NET] UDP bind has been updated
2022-05-13 18:21:56.842925: [NET] peer(CsqG…0myU) - Sending keepalive packet
2022-05-13 18:21:56.842957: [NET] Routine: receive incoming v4 - started
2022-05-13 18:21:56.843044: [NET] Routine: receive incoming v6 - started
2022-05-13 18:21:59.605094: [NET] Network change detected with satisfied route and interface order [pdp_ip0, utun2]
2022-05-13 18:21:59.606255: [NET] DNS64: mapped 93.149.216.183 to itself.
2022-05-13 18:21:59.606651: [NET] peer(CsqG…0myU) - UAPI: Updating endpoint
2022-05-13 18:21:59.607020: [NET] Routine: receive incoming v4 - stopped
2022-05-13 18:21:59.607216: [NET] Routine: receive incoming v6 - stopped
2022-05-13 18:21:59.607648: [NET] UDP bind has been updated
2022-05-13 18:21:59.607734: [NET] Routine: receive incoming v4 - started
2022-05-13 18:21:59.607781: [NET] peer(CsqG…0myU) - Sending keepalive packet
2022-05-13 18:21:59.607846: [NET] Routine: receive incoming v6 - started

2022-05-13 18:23:22.745841: [NET] peer(CsqG…0myU) - Receiving keepalive packet
2022-05-13 18:23:47.588631: [NET] peer(CsqG…0myU) - Receiving keepalive packet
2022-05-13 18:25:49.853450: [NET] peer(CsqG…0myU) - Sending handshake initiation
2022-05-13 18:25:50.185792: [NET] peer(CsqG…0myU) - Received handshake response
2022-05-13 18:26:05.228273: [NET] peer(CsqG…0myU) - Retrying handshake because we stopped hearing back after 15 seconds
(...)
```



> If you just turn off and turn back on the VPN when it stops working, does it start working again?

Unfortunately it does not.

**Comment by Raizelian** [comment]
Oh, thank you! :) I didn't know that, I'm still new at the whole GitHub thing. 

I went all the way down to an MTU value of 1000 but got nothing. Should I keep trying with even lower values?

**Comment by orazioedoardo** [comment]
> Should I keep trying with even lower values?

No just turn it back on to automatic. Lowering the MTU can fix some issues with networks that are unable to carry bigger packets, but 1000 is quite low.

Can you summarize what works all the time - only occasionally - never, when doing the following tests, possibly **trying different networks too**?

1) Ping 10.23.80.1
2) Ping 192.168.1.1
3) Load http://192.168.1.1
4) Load some websites.
5) Downloading 100 MB or so( to check how stable it is).

**Comment by Raizelian** [comment]
1. Never (only worked once)
2. Never (only worked once) 
3. Never (only worked once)
4. Never (only worked once)
5. Never tried it and the VPN never worked after that one time it did so I can’t even test it right now

1 to 4 all refer to the same instance that happened in the Wireguard logs on my iPhone that I shared a few posts prior to this one.

With trying different networks you mean using different clients? I tried Wireguard on Windows a few hours ago but it still didn’t work.

**Comment by orazioedoardo** [comment]
> With trying different networks you mean using different clients? I tried Wireguard on Windows a few hours ago but it still didn’t work.

No a network like public wifi, friend’s network, cellular network.

**Comment by Raizelian** [comment]
I tried a bit of everything and anything, but nothing worked.
I even tried setting up Wireguard manually by following a very in-depth guide (not sure if I can link to it, but [here it is](https://www.sigmdel.ca/michel/ha/wireguard/wireguard_02_en.html) - if it needs to be removed I'll do it) but it just wouldn't work at all.

Then I tried reconfiguring pivpn and setting up OpenVPN, and that worked right out of the box. I'm able to connect to the internet and pinging my local machines and my router works, too.
Can I grab some info from how it's working and use that info to try a new Wireguard configuration, you think?

**Comment by orazioedoardo** [comment]
> I even tried setting up Wireguard manually by following a very in-depth guide (not sure if I can link to it, but [here it is](https://www.sigmdel.ca/michel/ha/wireguard/wireguard_02_en.html) - if it needs to be removed I'll do it) but it just wouldn't work at all.

There isn't much room for maneuver with wireguard itself, I think this is something that has to do with the network, that's why I've asked you to try another.

> Can I grab some info from how it's working and use that info to try a new Wireguard configuration, you think?

No, openvpn and wireguard are fundamentally different.

**Comment by williamgh2019** [comment]
Same my problem. No internet connect if I connect to PiVPN. Before it working normal after that I no any change configure but after few months it suddenly no internet accesses if I connect PiVPN. Now change to use OpenVN by PIVPN it working. Pls check WireGuard in PiPVN
